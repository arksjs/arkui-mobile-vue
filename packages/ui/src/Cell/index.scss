@import '../style/var.scss';

.ta-cell {
  --ta-label-color: #{$title-color};
  --ta-content-color: #{$font-color};

  display: block;
  min-height: 48px;
  background-color: $white-color;
  position: relative;
  width: 100%;

  &:not(:last-child)::after {
    content: '';
    margin-left: 16px;
    width: calc(100% - 16px);
  }

  &.has--icon::after {
    margin-left: 48px;
    width: calc(100% - 48px);
  }

  &_header {
    padding: 8px 16px;
    box-sizing: border-box;
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
  }

  &_content {
    flex-grow: 1;
    flex-shrink: 0;
    font-size: 17px;
    color: var(--ta-content-color);
    min-height: 32px;
    line-height: 32px;
    display: flex;
    align-items: center;
    max-width: 100%;
  }

  &_label {
    flex-shrink: 0;
    height: 32px;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    line-height: 32px;
    color: var(--ta-label-color);
    font-size: 17px;
    padding-right: 12px;

    + .ta-cell_content {
      justify-content: flex-end;
    }
  }

  &_icon {
    margin: 0 12px 0 0;

    > .ta-icon {
      --ta-icon-size: 20px;
      --ta-icon-color: var(--ta-label-color);
    }
  }

  &_link-icon {
    margin-left: 4px;

    --ta-icon-size: 20px;
    --ta-icon-color: var(--ta-content-color);

    &.arrow {
      &--down {
        transform: rotate(90deg);
      }

      &--left {
        transform: rotate(180deg);
      }

      &--up {
        transform: rotate(270deg);
      }
    }
  }

  &_body {
    width: 100%;
    line-height: 20px;
    font-size: 14px;
    color: $font2-color;
    margin: -8px 0 0 16px;
    padding-bottom: 12px;
    word-break: break-all;
    word-wrap: break-word;
    overflow: hidden;
    text-align: left;

    .ta-cell.has--icon & {
      margin-left: 48px;
    }
  }

  &:not(.disabled).clickable {
    cursor: pointer;

    &:active {
      background-color: $background-color;
    }
  }

  &.disabled {
    --ta-content-color: #{rgba(var(--ta-grey-8), 0.2)};

    .ta-cell {
      &_label,
      &_body {
        opacity: 0.2;
      }
    }

    cursor: not-allowed;
  }

  .ta {
    &-input {
      height: 32px;
      border-width: 0;
      padding: 4px 0;
      border-radius: 0;

      &_input {
        text-align: right;
      }

      &.disabled {
        background-color: transparent;
      }
    }

    &-textarea {
      height: 92px;
    }

    &-radio-group {
      justify-content: flex-end;

      &.vertical .ta-radio_box {
        justify-content: flex-end;
      }
    }

    &-checkbox-group {
      justify-content: flex-end;

      &.vertical .ta-checkbox_box {
        justify-content: flex-end;
      }
    }

    &-picker-view {
      margin: -8px 0;
    }

    &-image-uploader {
      padding: 0;
      width: 192px;
      margin-left: auto;

      &.column-2 {
        width: 128px;
      }

      &.column-1 {
        width: 64px;
      }
    }
  }
}
