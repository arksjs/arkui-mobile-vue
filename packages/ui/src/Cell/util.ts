import { getEnumsValue } from '../helpers'
import type { ArrowDirection, CellProps } from './types'

export const LINK_ICON_NAMES: ArrowDirection[] = ['right', 'up', 'down', 'left']

export const getCellClasses = (props: CellProps) => {
  return [
    'ta-cell',
    'ta-horizontal-hairline',
    {
      clickable: !!(props.clickable || props.isLink),
      'has--icon': props.icon,
      disabled: props.disabled
    }
  ]
}

export const getCellArrowClasses = (arrowDirection?: ArrowDirection) => {
  return ['ta-cell_link-icon', 'arrow--' + getEnumsValue(LINK_ICON_NAMES, arrowDirection)]
}
