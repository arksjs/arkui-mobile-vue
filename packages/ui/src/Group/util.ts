export function getClasses(strongHeader: boolean) {
  return [
    ['ak-group', 'ak-horizontal-hairline'],
    { 'strong-header': strongHeader }
  ]
}
