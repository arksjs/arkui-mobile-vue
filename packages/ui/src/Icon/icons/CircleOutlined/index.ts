export { default as default } from './CircleOutlined.vue'
