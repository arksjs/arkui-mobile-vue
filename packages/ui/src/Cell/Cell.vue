<template>
  <div :class="classes" @click="onClick">
    <div class="ak-cell_header">
      <div class="ak-cell_icon" v-if="$slots.icon">
        <slot name="icon"></slot>
      </div>
      <div class="ak-cell_icon" v-else-if="icon"><Icon :icon="icon" /></div>
      <div class="ak-cell_label" v-if="label">
        {{ label }}
        <span class="ak-cell_required" v-if="required">*</span>
      </div>
      <div class="ak-cell_content">
        <slot>{{ content }}</slot>
      </div>
      <Icon v-if="isLink" :class="arrowClasses" :icon="RightOutlined" />
    </div>
    <div class="ak-cell_body" v-if="description">
      {{ description }}
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import type { PropType } from 'vue'
import { Icon } from '../Icon'
import {
  createEnumsValidator,
  emitClickValidator,
  iconValidator
} from '../helpers/validator'
import type { ArrowDirection, CellEmits } from './types'
import RightOutlined from '../Icon/icons/RightOutlined'
import type { IconData } from '../Icon/types'
import { getCellArrowClasses, getCellClasses, LINK_ICON_NAMES } from './util'
import type { OnClick, PropsToEmits } from '../helpers/types'

export default defineComponent({
  name: 'ak-cell',
  components: { Icon },
  props: {
    icon: {
      type: [String, Object] as PropType<IconData>,
      validator: iconValidator
    },
    label: {
      type: String,
      default: ''
    },
    description: {
      type: String,
      default: ''
    },
    content: {
      type: String,
      default: ''
    },
    clickable: {
      type: Boolean,
      default: false
    },
    required: {
      type: Boolean,
      default: false
    },
    isLink: {
      type: Boolean,
      default: false
    },
    arrowDirection: {
      type: String as PropType<ArrowDirection>,
      validator: createEnumsValidator(LINK_ICON_NAMES)
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },
  emits: {
    click: emitClickValidator
  } as PropsToEmits<CellEmits>,
  setup(props, { emit }) {
    const onClick: OnClick = e => {
      if (!props.disabled) {
        emit('click', e)
      }
    }

    const classes = computed(() => getCellClasses(props))
    const arrowClasses = computed(() =>
      getCellArrowClasses(props.arrowDirection)
    )

    return {
      classes,
      arrowClasses,
      RightOutlined,
      onClick
    }
  }
})
</script>
