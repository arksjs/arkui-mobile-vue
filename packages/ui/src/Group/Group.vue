<template>
  <div :class="classes">
    <div class="fx-group_header">
      <div class="fx-group_title">{{ title }}</div>
      <div class="fx-group_more">
        <slot name="header"></slot>
      </div>
    </div>
    <div class="fx-group_body fx-horizontal-hairline hairline-reverse">
      <div class="fx-group_body-inner">
        <slot></slot>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import { getClasses } from './util'

export default defineComponent({
  name: 'fx-group',
  props: {
    title: {
      type: String,
      required: true
    },
    strongHeader: {
      type: Boolean,
      default: false
    }
  },
  setup(props) {
    const classes = computed(() => getClasses(props.strongHeader))

    return {
      classes
    }
  }
})
</script>
