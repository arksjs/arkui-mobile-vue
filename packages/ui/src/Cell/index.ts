import { withInstall } from '../helpers/with-install'
import Cell from './Cell.vue'

export { Cell }
export const install = withInstall(Cell)
export default Cell
