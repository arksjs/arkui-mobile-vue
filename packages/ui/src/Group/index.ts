import { withInstall } from '../helpers/with-install'
import Group from './Group.vue'

export { Group }
export const install = withInstall(Group)
export default Group
