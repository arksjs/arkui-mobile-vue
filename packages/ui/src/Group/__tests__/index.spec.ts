import { describe, test, expect } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import { Group } from '@/Group'
import { Cell } from '@/Cell'
import CircleOutlined from '@/Icon/icons/CircleOutlined'

const title = 'title'

describe('Group', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(Group, {
      props: {
        title
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('title', () => {
    const wrapper = shallowMount(Group, {
      props: {
        title
      }
    })

    expect(wrapper.find('.ak-group_title').text()).toBe(title)
  })

  test('strongHeader', () => {
    const wrapper = shallowMount(Group, {
      props: {
        title,
        strongHeader: true
      }
    })

    expect(wrapper.classes()).toContain('strong-header')
  })

  test('slots', () => {
    const wrapper = mount(Group, {
      props: { title },
      slots: {
        header: CircleOutlined,
        default: [Cell]
      }
    })

    expect(wrapper.findComponent(CircleOutlined).exists()).toBeTruthy()
    expect(wrapper.findComponent(Cell).exists()).toBeTruthy()
  })
})
