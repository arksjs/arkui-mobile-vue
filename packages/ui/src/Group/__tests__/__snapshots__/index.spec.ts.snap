// Vitest Snapshot v1

exports[`Group > snapshot 1`] = `
"<div class=\\"ta-group ta-horizontal-hairline\\">
  <div class=\\"ta-group_header\\">
    <div class=\\"ta-group_title\\">title</div>
    <div class=\\"ta-group_more\\"></div>
  </div>
  <div class=\\"ta-group_body ta-horizontal-hairline hairline-reverse\\">
    <div class=\\"ta-group_body-inner\\"></div>
  </div>
</div>"
`;
